### PR TITLE
[scalardl-auditor] Support TLS in ScalarDL Auditor chart

### PR DESCRIPTION
**Pull request by kota2and3kan**
## Description

This PR adds TLS support in the ScalarDL Auditor chart.

In this update, users can configure TLS features and set key/cert files for ScalarDL Auditor.

Note: You need to enable the wire encryption feature on the ScalarDL Auditor side by setting `scalar.dl.auditor.server.tls.enabled=true` and `scalar.dl.auditor.tls.enabled=true` in the `auditor.auditorProperties` in your custom values file.

```console
                         +---(TLS)---> [ScalarDL Ledger] <---(TLS)------------------+
                         |                                                          |
[Client] ---> [Envoy] ---+                                                          |
                         |                                                          |
                         +---(TLS)---> [ScalarDL Auditor] ---(Recovery operation)---+
```

## Related issues and/or PRs

* https://github.com/scalar-labs/scalar-envoy/pull/28
  * I updated the Scalar Envoy to support TLS both downstream and upstream connections.
* https://github.com/scalar-labs/helm-charts/pull/253
  * I updated the Helm Chart of Scalar Envoy to support TLS between Envoy and ScalarDL Auditor.
* https://github.com/scalar-labs/docs-internal-orchestration/pull/14
  * I updated the document that is related to this PR. You can see the usage of this new feature and the getting started guide.

## Changes made

* Add new values to configure the TLS configurations.
* Mount key/cert file to use TLS connections between client (Envoy) and ScalarDL Auditor in the Deployment manifest.
* Mount key/cert file to use TLS connections between ScalarDL Auditor and ScalarDL Ledger in the Deployment manifest.

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

N/A

## Release notes

Support TLS configuration in the ScalarDL Auditor chart. You can enable TLS in both "Envoy - ScalarDL Auditor" and "ScalarDL Auditor - ScalarDL Ledger" connections.

### CODE REVIEW

**Review by supl — APPROVED** [review]
LGTM.
Thank you 🙇

**Review by choplin — APPROVED** [review]
LGTM!

File: charts/scalardl-audit/templates/auditor/deployment.yaml [hunk 1]
```diff
@@ -122,13 +162,27 @@ spec:
               command:
               - /usr/local/bin/grpc_health_probe
               - -addr=:40051
+              {{- if .Values.auditor.tls.enabled }}
+              - -tls
+              {{- if .Values.auditor.tls.caRootCertSecret }}
+              - -tls-ca-cert=/tls/certs/ca-root-cert.pem
+              {{- end }}
+              - -tls-server-name={{ .Values.auditor.tls.overrideAuthority }}
```
Comment (feeblefakie) [inline charts/scalardl-audit/templates/auditor/deployment.yaml:170]: Ditto.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you! 
Left one question for clarification.

File: charts/scalardl-audit/templates/auditor/deployment.yaml (hunk 1 above)
Comment (kota2and3kan) [reply charts/scalardl-audit/templates/auditor/deployment.yaml:171]: Thank you for your review!
I answered it on the ScalarDB Cluster PR side.
https://github.com/scalar-labs/helm-charts/pull/255#discussion_r1535123060

**Review by jnmt — APPROVED** [review]
LGTM! Thank you! (Please check CI issues, just in case.)

**Pull request merged**